src/services/transaction/TransactionMaterialService.ts

```diff
@@ -11,7 +11,8 @@ export class TransactionMaterialService extends AbstractService {
 	 */
 	async fetchTransactionMaterial(
 		hash: BlockHash,
-		noMeta: boolean
+		noMeta: boolean,
+		decodeMeta: boolean
```

Comment:
Just wondering; APi wise I guess we want to avoid breaking changes where possible, but I wonder about combining `decodeMeta` and  `noMeta` into one param that would be used like:

- `/?metadata=json` - return metadata decoded as JSON.
- `/?metadata=scale` - return the scale encoded blob.
- If `metadata` is no provided, don't return any metadata.

Just because at the mo you can provide combinations that don't make sense like `?noMeta&decodeMeta`.

Just a thought :)

Reply:
This is a great idea, and an enhancement for sure. The one issue being that it would introduce a breaking change, but I think it's justifiable, and cleaner. 

@emostov Any thoughts on this?

Reply:
So I discussed it in the integration channel, and I am going to go forward with your idea and implement `metadata` query param. The only thing being that we want to give users ample time to update because of how often we release sidecar. So I am going to have 2 params here. One is `noMeta`, and the other is `metadata`. Well allow users to migrate over something like 2 releases, then remove `noMeta` completely

Reply:
this sounds good to me